tests/run_compiled_model_hotswap.py

```diff
@@ -0,0 +1,60 @@
+# Copyright 2024-present the HuggingFace Inc. team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
```

Comment:
Would it also make sense to have a similar script for `diffusers`, too?

Reply:
Yeah but here we're only testing models from `transformers`.

Reply:
If I wanted to add a similar script but for diffusers, it would be a copy of the mentioned script. Should we have the same test twice?

Reply:
I think yes following the Stable Diffusion tests we have in `peft`.

Reply:
I added the diffusion model test but it fails on CI. At first I was puzzled, since it works locally, but in the end it's not surprising. The error is:

```
  + V1018 13:06:28.913644 140564106021760 torch/_dynamo/guards.py:2148] [0/0] [__guards] 
  + Traceback (most recent call last):
  +   File "/home/runner/work/peft/peft/tests/run_compiled_diffusion_model_hotswap.py", line 141, in <module>
  +     check_hotswap(do_hotswap=sys.argv[1] == "1")
  +   File "/home/runner/work/peft/peft/tests/run_compiled_diffusion_model_hotswap.py", line 129, in check_hotswap
  +     unet.load_attn_procs(file_name, adapter_name="default_0", hotswap=True)
  +   File "/opt/hostedtoolcache/Python/3.8.18/x64/lib/python3.8/site-packages/huggingface_hub/utils/_validators.py", line 114, in _inner_fn
  +     return fn(*args, **kwargs)
  +   File "/opt/hostedtoolcache/Python/3.8.18/x64/lib/python3.8/site-packages/diffusers/loaders/unet.py", line 206, in load_attn_procs
  +     is_model_cpu_offload, is_sequential_cpu_offload = self._process_lora(
  +   File "/opt/hostedtoolcache/Python/3.8.18/x64/lib/python3.8/site-packages/diffusers/loaders/unet.py", line 303, in _process_lora
  +     raise ValueError(
  + ValueError: Adapter name default_0 already in use in the Unet - please select a new adapter name.
```

This is expected because diffusers does not implement the hotswap feature yet. Locally, I worked with that branch, that's why it passed. Therefore, we can't include the test until diffusers adds the feature.

Since we can't know for sure which diffusers version will include the feature, I'd probably go with marking the whole test as `xfail` for now. WDYT?

Reply:
Yeah that works for me!

Reply:
Done: https://github.com/huggingface/peft/pull/2120/files#diff-df9ecc7077bee932f56e76161ada47693d73acd3ed175a5b9a9158cfe03ec381R1815